app/controllers/search_goods.js

```diff
@@ -0,0 +1,92 @@
+import Ember from "ember";
+import searchModule from "./search_module";
+import _ from "lodash";
+
+export default searchModule.extend({
+  minSearchTextLength: 2,
+  filteredResults: [],
+  displayResults: false,
+
+  onSearchTextChange: Ember.observer("searchText", function() {
+    if (this.get("searchText").length > this.get("minSearchTextLength")) {
+      this.reloadResults();
+    }
+    this.set("filteredResults", []);
+  }),
+
+  reloadResults() {
+    this.hideResults();
+    Ember.run.debounce(this, this.showResults, 500);
+  },
+
+  hideResults() {
+    Ember.run(() => {
+      this.set("displayResults", false);
+    });
+  },
+
+  showResults() {
+    Ember.run(() => {
+      this.set("displayResults", true);
+    });
+  },
+
+  getFilterQuery() {
+    return {
+      state: "in_stock,published"
+    };
+  },
+
+  getSearchQuery() {
+    return {
+      searchText: this.get("searchText"),
+      browseSearch: true
```

Comment:
@abhinav098  could you share what that property does and why we need it ?

Reply:
never mind, saw the api pull request and added a comment there : )

Reply:
yep, understood, and removed the property